warp/yul/ExecEnv.py

```diff
@@ -0,0 +1,233 @@
+from __future__ import annotations
```

Comment:
There is `CallGraphBuilder` (and `cleanup_callgraph`) in `top_sort.py`, did you look into whether it can be reused?